wire-schema/src/main/java/com/squareup/wire/schema/internal/parser/ProtoParser.java

```diff
@@ -275,18 +245,17 @@ private ExtendElement readExtend(Location location, String documentation) {
 
   /** Reads a service declaration and returns it. */
   private ServiceElement readService(Location location, String documentation) {
-    String name = readName();
+    String name = p.readName();
     ServiceElement.Builder builder = ServiceElement.builder(location)
         .name(name)
         .documentation(documentation);
 
-    if (readChar() != '{') throw unexpected("expected '{'");
+    if (p.readChar() != '{') throw p.unexpected("expected '{'");
```

Comment:
This pattern happens 100s of times. Maybe make an `expectChar(char)`, `expectWord(String)`, etc. set of methods inside the parser.